js.js

```diff
@@ -0,0 +1,137 @@
+//Losuje liczby z wybranego parametrami przedzialu
+function fromRange(min, max) {
+  return Math.floor(Math.random() * (max - min) + min);
+}
+
+function Game(board /* HTMLElement */) {
+  this.interval = null;
+  this.delay = 5000;
+  this.board = board;
+  this.lifes = 5;
+  this.point = 0;
+  this.timer = 5;
+  let time = document.querySelector("#time");
+  //odliczanie czasu
+  this.clock = setInterval(() => {
+    time.innerHTML = "czas " + this.timer;
+    this.timer--;
+    if (this.timer == 0) {
+      this.timer = 5;
+    }
+  }, 1000);
+  //zatrzymuje odliczanie czasu
+  this.stopClock = function() {
+    clearInterval(this.clock);
+  };
+  //klikniecie na board
+  this.board.addEventListener("click", () => {
+    let life = document.querySelector("#life");
+
+    if (this.lifes <= 0) {
+      this.stop();
+      this.stopClock();
+      this.board.removeEventListener("click");
+    }
+    this.lifes -= 1;
+    life.innerHTML = "życie " + this.lifes;
+  });
+}
+
+Game.prototype = {
+  start: function() {
+    this.interval = setInterval(() => {
+      const s = new Square();
+      this.board.appendChild(s.render());
+      //gdy jest wiecęj 5 kwadratow to zatrzymaj
+      if (this.board.children.length >= 5) {
+        //zatrzymuje odliczanie czasu gdy jest 5 kwadratów na planszy
+        this.stopClock();
+        this.stop();
+      }
+
+      //użyć bind albo arrow function
+      s.on("getPoint", () => {
+        this.point += 1;
+        let points = document.querySelector("#points");
+        points.innerHTML = "punkty " + this.point;
+      });
+      s.off("getPoint", () => {
```

Comment:
hm... to nie dziala... w sensie off nie odpina ...
zeby odpiac musisz przekazac dokladnie ta sama funkcje (jej referencje) a nie funkcje ktora wyglada tak samo.

Reply:
czyli powinno byc:
```javascript
const onPoint = () => {
/* ... */
// i na przyklad w srodku s.off("getPoint", onPoint);
};
s.on("getPoint", onPoint)